File: jan3/jan3.js
function judgeCircle(moves) {
    let x = 0, y = 0;
    for (let move of moves) {
        if (move === 'U') y++;
        else if (move === 'D') y--;
        else if (move === 'L') x--;
        else if (move === 'R') x++;
    }
    return x === 0 && y === 0;
}

const input = "UD";
const result = judgeCircle(input);
console.log(result);

File: jan3/jan3.rb
def judge_circle(moves)
    x = 0
    y = 0
    moves.each_char do |move|
        case move
        when 'U'
            y += 1
        when 'D'
            y -= 1
        when 'L'
            x -= 1
        when 'R'
            x += 1
        end
    end
    x.zero? && y.zero?
end

input = "UD"
result = judge_circle(input)
puts result
